src/lib.rs

```diff
@@ -658,6 +776,13 @@ mod tests {
         cmp_content(out_file, "This is a test!");
     }
 
+    #[cfg(not(feature = "archive-tar"))]
+    #[test]
+    #[ignore]
+    fn unpack_tar_gzip() {
+        println!("WARNING: Please enable 'archive-tar' feature!");
+    }
```

Comment:
do we need to keep these "ignored" tests around?

Reply:
I though those tests could still be included and ignored by default. This results in the following output which I think is nice:

```text
❯ cargo test
    Finished test [unoptimized + debuginfo] target(s) in 0.11s
     Running target/debug/deps/self_update-ea394c90501341d2

running 17 tests
test tests::detect_plain_tar ... ignored
test tests::detect_tar_gz ... ignored
test tests::detect_zip ... ignored
test tests::unpack_file_plain_gzip ... ignored
test tests::unpack_file_tar_gzip ... ignored
test tests::unpack_plain_gzip ... ignored
test tests::unpack_plain_gzip_double_ext ... ignored
test tests::unpack_tar_gzip ... ignored
test tests::unpack_zip ... ignored
test tests::unpack_zip_file ... ignored
test tests::detect_plain ... ok
test tests::detect_plain_gz ... ok
test version::test::test_bump_greater ... ok
test version::test::test_bump_is_major ... ok
test version::test::test_bump_is_patch ... ok
test version::test::test_bump_is_compatible ... ok
test version::test::test_bump_is_minor ... ok

test result: ok. 7 passed; 0 failed; 10 ignored; 0 measured; 0 filtered out

   Doc-tests self_update

running 5 tests
test src/lib.rs -  (line 82) ... ok
test src/lib.rs -  (line 38) ... ok
test src/backends/s3.rs - backends::s3::UpdateBuilder::bin_path_in_archive (line 239) ... ok
test src/backends/github.rs - backends::github::UpdateBuilder::bin_path_in_archive (line 312) ... ok
test src/lib.rs -  (line 62) ... ok

test result: ok. 5 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```

versus:

```text
❯ cargo test --all-features
   Compiling self_update v0.11.1 (/Users/nbigaouette/codes/rust/self_update.git)
    Finished test [unoptimized + debuginfo] target(s) in 7.03s
     Running target/debug/deps/self_update-02b5550d4a8a427c

running 17 tests
test tests::detect_zip ... ok
test tests::detect_plain_gz ... ok
test tests::detect_plain ... ok
test tests::detect_tar_gz ... ok
test tests::detect_plain_tar ... ok
test version::test::test_bump_greater ... ok
test version::test::test_bump_is_compatible ... ok
test version::test::test_bump_is_major ... ok
test version::test::test_bump_is_minor ... ok
test version::test::test_bump_is_patch ... ok
test tests::unpack_file_plain_gzip ... ok
test tests::unpack_plain_gzip_double_ext ... ok
test tests::unpack_plain_gzip ... ok
test tests::unpack_zip_file ... ok
test tests::unpack_zip ... ok
test tests::unpack_file_tar_gzip ... ok
test tests::unpack_tar_gzip ... ok

test result: ok. 17 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

   Doc-tests self_update

running 5 tests
test src/backends/s3.rs - backends::s3::UpdateBuilder::bin_path_in_archive (line 239) ... ok
test src/lib.rs -  (line 62) ... ok
test src/backends/github.rs - backends::github::UpdateBuilder::bin_path_in_archive (line 312) ... ok
test src/lib.rs -  (line 38) ... ok
test src/lib.rs -  (line 82) ... ok

test result: ok. 5 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```

This way it is clear that something is not activated. One can still run the ignored tests using `cargo test -- --ignored`. But then I realized that I am using a `printl!()` when I wanted to use a `panic!()` instead (to fail the tests).

That was my reasoning anyway. What would you prefer to see here?

Reply:
Ok, I think it makes sense how you've done it 👍